plugins/transformations/seq2array/srcCxx/Seq2ArrayTransformation.cxx

```diff
@@ -0,0 +1,481 @@
+            // (output with arrays and input with sequences)
+            switch (input_sequence_type.content_type().kind().underlying()) {
+                case TypeKind::STRUCTURE_TYPE:
+                case TypeKind::SEQUENCE_TYPE:
+                case TypeKind::ARRAY_TYPE:
+                case TypeKind::UNION_TYPE:
+                    is_compatible = is_compatible && are_type_compatible(
+                            input_sequence_type.content_type(),
+                            output_array_type.content_type());
+                    if (!is_compatible) {
+                        std::string error("Error: content of input member <"
+                                + input_sequence_type.name()
+                                + ">, not compatible with content of output member <"
+                                + output_array_type.name()
+                                + ">");
+                        throw std::runtime_error(error);
+                    }
+                break;
+                // primitive type
+                default:
+                    // not needed because if it is a primitive content_type, this
+                    // has been already checked in the previous if condition
+                    break;
+                }
+            }
+            break;
+        }
+    default:
+        std::string error("Error: incompatible type input <" + input_type.name()
+                + ">, and output <" + output_type.name() + ">");
+        throw std::runtime_error(error);
+        break;
+    }
+
+    return is_compatible;
+}
+
+Seq2ArrayTransformation::Seq2ArrayTransformation(
+        const rti::routing::TypeInfo &input_type_info,
+        const rti::routing::TypeInfo &output_type_info,
+        const rti::routing::PropertySet &properties)
+        : input_type_info_(input_type_info.dynamic_type()),
+          output_type_info_(output_type_info.dynamic_type())
+{
+    if (!are_type_compatible(input_type_info_, output_type_info_)) {
+        throw std::runtime_error("Error: input and ouput types are not compatible");
+    }
+    // properties are not used because there is no additional configuration for
+    // this transformation
+}
+
+/*
+ * @brief Substitutes sequences by arrays of any type.
+ *
+ */
+void Seq2ArrayTransformation::convert_sample(
+        DynamicData &input_sample,
+        DynamicData &output_sample)
+{
+    // convert all the elements within a struct, sequence, array or union.
+    for (uint32_t i = 1; i <= input_sample.member_count(); ++i) {
+        uint32_t member_to_process = i;
+        bool is_union_member = false;
+
+        // unions only have to check the value that the discriminator identifies
+        if (input_sample.type_kind() == TypeKind::UNION_TYPE) {
+            member_to_process = input_sample.discriminator_value();
+            is_union_member = true;
+        }
+
+        if (!input_sample.member_exists(member_to_process)) {
+            continue;
+        }
```

Comment:
How is this possible? Do optional members get reported in the "member count" but don't "exist"?

Reply:
I thought that's possible, I need to test that

Reply:
I've tested it and that's correct